src/main/treeDataProviders/utils/descriptorUtils.ts

```diff
@@ -146,6 +153,28 @@ export class DescriptorUtils {
         return descriptorNode.dependenciesWithIssue.length;
     }
 
+    /**
+     * Convert descriptor impacted paths to a set of direct components that are impacted.
+     * The first children are the direct components.
+     * @param impactedPaths - the impacted path to convert
+     * @returns set of direct components in the impacted path
+     */
+    public static getDirectComponents(impactedPaths: Map<string, IImpactedPath>): Set<string> {
+        let result: Set<string> = new Set<string>();
+
+        for (const impactedPath of impactedPaths.values()) {
+            if (impactedPath.children) {
+                for (const directPath of impactedPath.children) {
+                    if (!result.has(directPath.name)) {
+                        result.add(directPath.name);
+                    }
```

Comment:
The Set data structure already checks for existence before adding an item
```suggestion
                    result.add(directPath.name);
```

Reply:
done